### PR TITLE
Async by default

### PR DESCRIPTION
**Pull request by mcollina**
Given that we are going to drop Node v12, we can actually be safe in enabling asynchronous logging everywhere.

This is based on top of #1249.

### CODE REVIEW

File: docs/asynchronous.md [hunk 1]
```diff
@@ -16,74 +12,19 @@ const logger = pino(pino.destination({
...
+In this mode of operation log messages are directly written to the
+output stream as the messages are generated with a _blocking_ operation.
+
 * See [`pino.destination`](/docs/api.md#pino-destination)
 * `pino.destination` is implemented on [`sonic-boom` ⇗](https://github.com/mcollina/sonic-boom).
 
 ### AWS Lambda
 
-On AWS Lambda we recommend to call `dest.flushSync()` at the end
+Asynchronous logging is disabled by default on AWS Lambda or any other environemnt
```
Comment (kibertoad) [inline docs/asynchronous.md:24]:
```suggestion
Asynchronous logging is disabled by default on AWS Lambda, or any other environment
```

File: docs/asynchronous.md [hunk 2]
```diff
@@ -16,74 +12,19 @@ const logger = pino(pino.destination({
...
+output stream, as the messages are generated with a _blocking_ operation.
+
 * See [`pino.destination`](/docs/api.md#pino-destination)
 * `pino.destination` is implemented on [`sonic-boom` ⇗](https://github.com/mcollina/sonic-boom).
 
 ### AWS Lambda
 
-On AWS Lambda we recommend to call `dest.flushSync()` at the end
+Asynchronous logging is disabled by default on AWS Lambda, or any other environment
+that modified `process.stdout`. If forcefully turned on, we recommend to call `dest.flushSync()` at the end
```
Comment (jsumners) [inline docs/asynchronous.md:25]:
```suggestion
that modifies `process.stdout`. If forcefully turned on, we recommend to call `dest.flushSync()` at the end
```

**Review by kibertoad — APPROVED** [review]

**Review by jsumners — APPROVED** [review]

**Comment by kibertoad** [comment]
Tests didn't need to be updated?

**Comment by mcollina** [comment]
No. Async and sync logging are functionally equivalent given the other semver-major changes.

**Pull request merged**